torch/distributed/_tensor/api.py

```diff
@@ -704,11 +704,11 @@ def replicate_module_params_buffers(m: nn.Module, mesh: DeviceMesh) -> None:
 
     # register input_fn as module forward pre hook
     if input_fn is not None:
-        module.register_forward_pre_hook(lambda _, inputs: input_fn(inputs, device_mesh))  # type: ignore[misc]
-    # register input_fn as module forward hook
+        module.register_forward_pre_hook(lambda mod, inputs: input_fn(mod, inputs, device_mesh))  # type: ignore[misc]
+    # register output_fn as module forward hook
```

Comment:
To be discussed:

Maybe we can check the passed in function num of args, and handle them correspondingly to avoid BC breaking, i.e. if `input_fn` have 3 args, then we pass module in, otherwise we don't

Need to see if that's better 🤔

Reply:
OK updated the PR to maintain BC, but also throwing deprecation warnings